driver-gpio-mpc8xxx-do-not-set-IRQ_TYPE_NONE

Apply upstream patch to avoid harmless, but annoying error messages
when setting GPIO via sysfs.

See commit de0ccf788147440eee2383c74408080f3ff0a43b in the stable
kernel tree.  Or this thread:

https://lkml.org/lkml/2013/2/4/77

diff --git a/drivers/gpio/gpio-mpc8xxx.c b/drivers/gpio/gpio-mpc8xxx.c
index 34be13b..60e259f 100644
--- a/drivers/gpio/gpio-mpc8xxx.c
+++ b/drivers/gpio/gpio-mpc8xxx.c
@@ -296,7 +296,6 @@ static int mpc8xxx_gpio_irq_map(struct irq_host *h, unsigned int virq,
 
 	irq_set_chip_data(virq, h->host_data);
 	irq_set_chip_and_handler(virq, &mpc8xxx_irq_chip, handle_level_irq);
-	irq_set_irq_type(virq, IRQ_TYPE_NONE);
 
 	return 0;
 }
